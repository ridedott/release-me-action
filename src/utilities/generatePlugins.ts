import { PluginSpec } from 'semantic-release';

export const generatePlugins = ({
  commitAssets,
  isNodeModule,
  releaseAssets,
}: {
  commitAssets: string[];
  isNodeModule: boolean;
  releaseAssets: string[];
}): PluginSpec[] => {
  return [
    '@semantic-release/commit-analyzer',
    '@semantic-release/release-notes-generator',
    '@semantic-release/changelog',
    [
      '@semantic-release/exec',
      {
        prepareCmd: 'npx prettier --write CHANGELOG.md',
      },
    ],
    ...(isNodeModule === true
      ? [
          [
            '@semantic-release/npm',
            {
              npmPublish: false,
            },
          ] as PluginSpec,
        ]
      : []),
    [
      '@semantic-release/git',
      {
        assets: [
          './CHANGELOG.md',
          ...commitAssets,
          ...(isNodeModule
            ? ['./package.json', './package-lock.json', './yarn-lock.yaml']
            : []),
        ],
        // eslint-disable-next-line no-template-curly-in-string
        message: 'chore(release): v${nextRelease.version}',
      },
    ],
    [
      '@semantic-release/github',
      {
        assets: releaseAssets,
        failComment: false,
        releasedLabels: false,
        successComment: false,
      },
    ],
  ];
};
