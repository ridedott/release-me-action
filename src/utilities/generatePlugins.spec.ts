import { generatePlugins } from './generatePlugins';

describe('generatePlugins', (): void => {
  it('excludes the npm plugin when called with the isNodeModule parameter set to false', (): void => {
    expect.assertions(1);

    expect(
      generatePlugins({
        commitAssets: [],
        isNodeModule: false,
        releaseAssets: [],
      }),
    ).toMatchSnapshot();
  });

  it('configures the git plugin to omit committing the updated package files when isNodeModule is set to false', (): void => {
    expect.assertions(1);

    expect(
      generatePlugins({
        commitAssets: [],
        isNodeModule: false,
        releaseAssets: [],
      }),
    ).toMatchSnapshot();
  });

  it('configures the npm plugin to bump the package versions when the isNodeModule parameter is set to true', (): void => {
    expect.assertions(1);

    expect(
      generatePlugins({
        commitAssets: [],
        isNodeModule: true,
        releaseAssets: [],
      }),
    ).toMatchSnapshot();
  });

  it('configures the git plugin to commit the updated package files when the isNodeModule parameter is set to true', (): void => {
    expect.assertions(1);

    expect(
      generatePlugins({
        commitAssets: [],
        isNodeModule: true,
        releaseAssets: [],
      }),
    ).toMatchSnapshot();
  });

  it('configures the git plugin to commit the specified assets as part of the release process', (): void => {
    expect.assertions(1);

    expect(
      generatePlugins({
        commitAssets: ['./src'],
        isNodeModule: false,
        releaseAssets: [],
      }),
    ).toMatchSnapshot();
  });

  it('configures the github plugin to release the specified assets as part of the release process', (): void => {
    expect.assertions(1);

    expect(
      generatePlugins({
        commitAssets: [],
        isNodeModule: false,
        releaseAssets: ['./src'],
      }),
    ).toMatchSnapshot();
  });
});
