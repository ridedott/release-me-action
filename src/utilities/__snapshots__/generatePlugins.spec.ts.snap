// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generatePlugins configures the git plugin to commit the specified assets as part of the release process 1`] = `
Array [
  "@semantic-release/commit-analyzer",
  "@semantic-release/release-notes-generator",
  "@semantic-release/changelog",
  Array [
    "@semantic-release/exec",
    Object {
      "prepareCmd": "npx prettier --parser markdown --write CHANGELOG.md",
    },
  ],
  Array [
    "@semantic-release/git",
    Object {
      "assets": Array [
        "./CHANGELOG.md",
        "./src",
      ],
      "message": "chore(release): v\${nextRelease.version}",
    },
  ],
  Array [
    "@semantic-release/github",
    Object {
      "assets": Array [],
      "failComment": false,
      "releasedLabels": false,
      "successComment": false,
    },
  ],
]
`;

exports[`generatePlugins configures the git plugin to commit the updated package files when the isNodeModule parameter is set to true 1`] = `
Array [
  "@semantic-release/commit-analyzer",
  "@semantic-release/release-notes-generator",
  "@semantic-release/changelog",
  Array [
    "@semantic-release/exec",
    Object {
      "prepareCmd": "npx prettier --parser markdown --write CHANGELOG.md",
    },
  ],
  Array [
    "@semantic-release/npm",
    Object {
      "npmPublish": false,
    },
  ],
  Array [
    "@semantic-release/git",
    Object {
      "assets": Array [
        "./CHANGELOG.md",
        "./package.json",
        "./package-lock.json",
        "./yarn-lock.yaml",
      ],
      "message": "chore(release): v\${nextRelease.version}",
    },
  ],
  Array [
    "@semantic-release/github",
    Object {
      "assets": Array [],
      "failComment": false,
      "releasedLabels": false,
      "successComment": false,
    },
  ],
]
`;

exports[`generatePlugins configures the git plugin to omit committing the updated package files when isNodeModule is set to false 1`] = `
Array [
  "@semantic-release/commit-analyzer",
  "@semantic-release/release-notes-generator",
  "@semantic-release/changelog",
  Array [
    "@semantic-release/exec",
    Object {
      "prepareCmd": "npx prettier --parser markdown --write CHANGELOG.md",
    },
  ],
  Array [
    "@semantic-release/git",
    Object {
      "assets": Array [
        "./CHANGELOG.md",
      ],
      "message": "chore(release): v\${nextRelease.version}",
    },
  ],
  Array [
    "@semantic-release/github",
    Object {
      "assets": Array [],
      "failComment": false,
      "releasedLabels": false,
      "successComment": false,
    },
  ],
]
`;

exports[`generatePlugins configures the github plugin to release the specified assets as part of the release process 1`] = `
Array [
  "@semantic-release/commit-analyzer",
  "@semantic-release/release-notes-generator",
  "@semantic-release/changelog",
  Array [
    "@semantic-release/exec",
    Object {
      "prepareCmd": "npx prettier --parser markdown --write CHANGELOG.md",
    },
  ],
  Array [
    "@semantic-release/git",
    Object {
      "assets": Array [
        "./CHANGELOG.md",
      ],
      "message": "chore(release): v\${nextRelease.version}",
    },
  ],
  Array [
    "@semantic-release/github",
    Object {
      "assets": Array [
        "./src",
      ],
      "failComment": false,
      "releasedLabels": false,
      "successComment": false,
    },
  ],
]
`;

exports[`generatePlugins configures the npm plugin to bump the package versions when the isNodeModule parameter is set to true 1`] = `
Array [
  "@semantic-release/commit-analyzer",
  "@semantic-release/release-notes-generator",
  "@semantic-release/changelog",
  Array [
    "@semantic-release/exec",
    Object {
      "prepareCmd": "npx prettier --parser markdown --write CHANGELOG.md",
    },
  ],
  Array [
    "@semantic-release/npm",
    Object {
      "npmPublish": false,
    },
  ],
  Array [
    "@semantic-release/git",
    Object {
      "assets": Array [
        "./CHANGELOG.md",
        "./package.json",
        "./package-lock.json",
        "./yarn-lock.yaml",
      ],
      "message": "chore(release): v\${nextRelease.version}",
    },
  ],
  Array [
    "@semantic-release/github",
    Object {
      "assets": Array [],
      "failComment": false,
      "releasedLabels": false,
      "successComment": false,
    },
  ],
]
`;

exports[`generatePlugins excludes the changelog plugin when called with the disableChangelog parameter set to true 1`] = `
Array [
  "@semantic-release/commit-analyzer",
  "@semantic-release/release-notes-generator",
  Array [
    "@semantic-release/git",
    Object {
      "assets": Array [],
      "message": "chore(release): v\${nextRelease.version}",
    },
  ],
  Array [
    "@semantic-release/github",
    Object {
      "assets": Array [],
      "failComment": false,
      "releasedLabels": false,
      "successComment": false,
    },
  ],
]
`;

exports[`generatePlugins excludes the npm plugin when called with the isNodeModule parameter set to false 1`] = `
Array [
  "@semantic-release/commit-analyzer",
  "@semantic-release/release-notes-generator",
  "@semantic-release/changelog",
  Array [
    "@semantic-release/exec",
    Object {
      "prepareCmd": "npx prettier --parser markdown --write CHANGELOG.md",
    },
  ],
  Array [
    "@semantic-release/git",
    Object {
      "assets": Array [
        "./CHANGELOG.md",
      ],
      "message": "chore(release): v\${nextRelease.version}",
    },
  ],
  Array [
    "@semantic-release/github",
    Object {
      "assets": Array [],
      "failComment": false,
      "releasedLabels": false,
      "successComment": false,
    },
  ],
]
`;
